crates/vfs-notify/src/lib.rs

```diff
@@ -85,10 +85,13 @@ impl NotifyActor {
     }
 
     fn next_event(&self, receiver: &Receiver<Message>) -> Option<Event> {
-        let watcher_receiver = self.watcher.as_ref().map(|(_, receiver)| receiver);
+        let Some((_, watcher_receiver)) = &self.watcher else {
+            return receiver.recv().ok().map(Event::Message);
+        };
+
         select! {
             recv(receiver) -> it => it.ok().map(Event::Message),
-            recv(watcher_receiver.unwrap_or(&never())) -> it => Some(Event::NotifyEvent(it.unwrap())),
+            recv(watcher_receiver) -> it => Some(Event::NotifyEvent(it.unwrap())),
```

Comment:
Why are we unwrapping here but not with other messages?

Reply:
Unsure, might've been due to drop order at some point to prevent a panic

Reply:
Well, not relevant for the PR. just noticed it.